CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp

```diff
@@ -456,32 +456,52 @@ void WeaponSystem::OnProjectileHit(const LambdaEngine::EntityCollisionInfo& coll
 	const ComponentArray<TeamComponent>*		pTeamComponents			= pECS->GetComponentArray<TeamComponent>();
 	const ComponentArray<ProjectileComponent>*	pProjectileComponents	= pECS->GetComponentArray<ProjectileComponent>();
 
+	// Detect selfhit
+	bool selfHit = false;
+	EAmmoType ammoType = EAmmoType::AMMO_TYPE_NONE;
+	if (pProjectileComponents->HasComponent(collisionInfo0.Entity))
+	{
+		const ProjectileComponent& projectilComp = pProjectileComponents->GetConstData(collisionInfo0.Entity);
+		ammoType = projectilComp.AmmoType;
+		
+		if (projectilComp.Owner == collisionInfo1.Entity)
+		{
+			selfHit = true;
+		}
+	}
+
+	// On selfhit return
+	if (selfHit)
+	{
+		LOG_INFO("SELF HIT");
+		return;
+	}
```

Comment:
Have you tested playing around with this? I could imagine the projctiles flying like crazy if they spawn inside the capsule and aren't deleted upon collision.

Reply:
Yes seems to work well